pubtools/pulplib/_impl/fake/match.py

```diff
@@ -124,6 +126,21 @@ def match_in(matcher, field, obj):
     return False
 
 
+@visit(LessThanMatcher)
+def match_less(matcher, field, obj):
+    # for datetime.dateime fields, obj_value is returned as pulp_value(iso format)
+    # if the model filed name is same as pulp filed name else returns a obj_value
+    # as datetime.datetime object
+    obj_value = get_field(field, obj)
```

Comment:
I guess this part will need refactoring per my suggestion that caller should be able to pass a datetime.datetime, not a str, when searching datetimes.

Reply:
Agree. Though this obj_value might need some conversion based on matcher._value type.